lib/intercom/request.rb

```diff
@@ -72,11 +72,11 @@ def execute(target_base_url = nil, token:, read_timeout: 90, open_timeout: 30, a
             parsed_body
           rescue Intercom::RateLimitExceeded => e
             if @handle_rate_limit
-              seconds_to_retry = (@rate_limit_details[:reset_at] - Time.now.utc).ceil
+              seconds_to_retry = ((@rate_limit_details[:reset_at] || Time.now.utc) - Time.now.utc).ceil
               if (retries -= 1) < 0
                 raise Intercom::RateLimitExceeded, 'Rate limit retries exceeded. Please examine current API Usage.'
               else
-                sleep seconds_to_retry unless seconds_to_retry < 0
+                sleep seconds_to_retry unless seconds_to_retry <= 0
```

Comment:
If you're keen to make the change then perhaps this?

```suggestion
                sleep seconds_to_retry if seconds_to_retry.positive?
```